src/shapes/audio_token.class.js

```diff
@@ -0,0 +1,202 @@
+(function(global) {
+
+  'use strict';
+
+  if (!global.fabric) {
+    global.fabric = { };
+  }
+
+  if (global.fabric.Audio_token) {
+    fabric.warn('fabric.Audio_token is already defined.');
+    return;
+  }
+
+  /**
+   * Audio_token class
+   * @class fabric.Audio_token
+   * @extends fabric.Object
+   * @see {@link fabric.Audio_token#initialize} for constructor definition
+   */
+  fabric.Audio_token = fabric.util.createClass(fabric.Object, /** @lends fabric.Audio_token.prototype */ {
+
+    /**
+     * Type of an object
+     * @type String
+     * @default
+     */
+    type: 'audio_token',
+
+    /**
+     * unique key - used for now just to confirm play/pause for specific tokens
+     * @since 2.0.0
+     * @type String
+     * @default
+     */
+    cacheKey: '',
+
+    /**
+     * Indicates whether audio is in 'play' mode
+     * @type Boolean
+     * @default
+     */
+    isPlaying: false,
+
+    /**
+     * Indicates whether audio is in 'paused' mode
+     * @type Boolean
+     * @default
+     */
+    isPaused: false,
+
+    /**
+     * Indicates where audio playback is currently (in ms) //not used yet
+     * @type number
+     * @default
+     */
+    playbackMS: 0,
+
+    /**
+     * Audio URL? WIP - not sure how this will be used
+     * To be determined by: NEWXP-1949, likely passed in as a parameter on creation
+     * @type string
+     * @default
+     */
+    audioUrl: '',
+
+    hasBorders: false,
+
+    /**
+     * extra space taken up on the left by delete controls (used for collision detection)
+     * @type Number
+     * @default
+     */
+    leftMargin: 44,
+
+    /**
+     * Definable images for idle state, selected state, and play/pause controls
```

Comment:
These are clunky. Currently no actual images live in fabricjs or amp-worksheet. I'm passing in hardcoded base64 pngs/strings. :(